### PR TITLE
ALEC-240 - Split agreement and engine rest endpoint logic

### PR DESCRIPTION
**Pull request by BenjaminJ**
Split agreement and engine endpoints to avoid agreement being not available if driver is not started

[alec-239](https://issues.opennms.org/browse/ALEC-240)

### CODE REVIEW

File: features/ui/src/main/java/org/opennms/alec/rest/ALECRestUtils.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.opennms.alec.rest;
```
Comment (amay94) [inline features/ui/src/main/java/org/opennms/alec/rest/ALECRestUtils.java:28]: Needs a copyright header

File: features/ui/src/test/java/org/opennms/alec/rest/EngineRestImplTest.java [hunk 2]
```diff
@@ -28,13 +28,9 @@
 import org.mockito.Spy;
```
Comment (amay94) [inline features/ui/src/test/java/org/opennms/alec/rest/EngineRestImplTest.java:55]: this class also needs a header

**Review by amay94 — CHANGES_REQUESTED** [review]
Looks good to me, other than the missing headers

File: features/ui/src/main/java/org/opennms/alec/rest/ALECRestUtils.java (hunk 1 above)
Comment (BenjaminJ) [reply features/ui/src/main/java/org/opennms/alec/rest/ALECRestUtils.java:28]: done

File: features/ui/src/test/java/org/opennms/alec/rest/EngineRestImplTest.java (hunk 2 above)
Comment (BenjaminJ) [reply features/ui/src/test/java/org/opennms/alec/rest/EngineRestImplTest.java:55]: done

**Review by aramosvizcarra — APPROVED** [review]
I would suggest to use DTO's instead of Responses. But since is a couple of classes and they are small enough is probably not necessary.
Otherwise Looks good!

**Review by amay94 — APPROVED** [review]

**Pull request merged**